pkg/envoy/lds/inmesh.go

```diff
@@ -274,9 +274,25 @@ func (lb *listenerBuilder) getOutboundFilterChainMatchForService(dstSvc service.
 		return nil, err
 	}
 
+	contains := func(s []string, t string) bool {
+		for _, a := range s {
+			if a == t {
+				return true
+			}
+		}
+		return false
+	}
+
+	var endpointSet []string
```

Comment:
I'd suggest, if it's a set, just use a real set (`mapset` maybe, since there's no standard set lib), will remove the need to keep your own linear search code.

Reply:
using mapset will later break ordering on the test. Instead of comparing ordered elements, we will keep this linear search which will produce deterministic slice order.